scio-avro/src/main/scala/com/spotify/scio/avro/AvroTypedIO.scala

```diff
@@ -61,6 +62,36 @@ object AvroTypedIO {
 @deprecated("Use AvroTypedIO instead", "0.14.0")
 object AvroTyped {
   type AvroIO[T <: HasAvroAnnotation] = AvroTypedIO[T]
-  def AvroIO[T <: HasAvroAnnotation: TypeTag: Coder](path: String): AvroIO[T] =
-    AvroTypedIO[T](path)
+  def AvroIO[T <: HasAvroAnnotation: TypeTag: Coder](path: String): AvroIO[T] = AvroTypedIO[T](path)
+}
+
+final case class AvroMagnolifyTyped[T: MagnolifyAvroType: Coder](path: String) extends ScioIO[T] {
```

Comment:
I think we should respect some naming conventions for our `ScioIO`s:
- start with file format
- end with IO

```suggestion
final case class AvroMagnolifyTypedIO[T: MagnolifyAvroType: Coder](path: String) extends ScioIO[T] {
```